### ISSUE TITLE
Fragment handling code doesn't invoke ExternalAcks::ack method

### ISSUE THREAD

**Issue by jstnlef** [amethyst/laminar#137]
In porting laminar to the new laminar-mio module, I was trying to get the fragmentation handling working correctly and noticed that the ExternalAcks record isn't getting updated when handling completed fragments. I tested this by updating the the following test methods in `VirtualConnection`.
`process_reliable_unordered_packet` and `process_fragmented_packet`.

```
fn process_reliable_unordered_packet() {
        let mut connection = create_virtual_connection();

        let buffer = vec![1; 500];

        for _ in 0..10 {
            let mut packet_data = connection
                .process_outgoing(&buffer, DeliveryMethod::ReliableUnordered)
                .unwrap();
            assert_eq!(packet_data.fragment_count(), 1);

            assert_packet_payload(&buffer, packet_data.parts(), &mut connection);
        }
        assert!(false);
    }
```

```
fn process_fragmented_packet() {
        let mut connection = create_virtual_connection();

        let buffer = vec![1; 4000];

        for _ in 0..10 {
            let mut packet_data = connection
                .process_outgoing(&buffer, DeliveryMethod::ReliableUnordered)
                .unwrap();

            // there should be 4 fragments
            assert_eq!(packet_data.fragment_count(), 4);

            for (index, part) in packet_data.parts().into_iter().enumerate() {
                let option = connection.process_incoming(&part).unwrap();

                // take note index 3 will contain the fragment data because the bytes of the fragmented packet will be returned when all fragments are processed.
                // that is why the last packet (index 3) can be asserted on.
                match option {
                    None => {
                        if index < 3 {
                            assert!(true)
                        } else {
                            assert!(false)
                        }
                    }
                    Some(packet) => {
                        if index == 3 {
                            assert_eq!(buffer, packet.payload());
                        }
                    }
                }
            }
        }
        assert!(false);
    }
```

I also updated `VirtualConnection::process_incoming` with the following print statement after making `their_acks` public.
```
if !ProtocolVersion::valid_version(header.protocol_version) {
            return Err(NetworkErrorKind::ProtocolVersionMismatch.into());
        }

        println!("Their acks before: {:?}", self.reliable_unordered_channel.their_acks);

        if header.packet_type_id == PacketTypeId::Fragment {
```

In the non-fragmented case, I get
```
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: true }
Their acks before: ExternalAcks { last_seq: 1, field: 1, initialized: true }
Their acks before: ExternalAcks { last_seq: 2, field: 3, initialized: true }
Their acks before: ExternalAcks { last_seq: 3, field: 7, initialized: true }
Their acks before: ExternalAcks { last_seq: 4, field: 15, initialized: true }
Their acks before: ExternalAcks { last_seq: 5, field: 31, initialized: true }
Their acks before: ExternalAcks { last_seq: 6, field: 63, initialized: true }
Their acks before: ExternalAcks { last_seq: 7, field: 127, initialized: true }
Their acks before: ExternalAcks { last_seq: 8, field: 255, initialized: true }
```
However, in the fragmented case, I get
```
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
Their acks before: ExternalAcks { last_seq: 0, field: 0, initialized: false }
```
(There are 40 lines since each packet has 4 fragments)

**Comment by TimonPost** [comment]
Closing this because it is fixed in #150

**Comment by jstnlef** [comment]
In the future, it's probably worth leaving these open until the PR is actually merged.
